### DISCUSSION TITLE
Reactivity on array on Flatlist in general

### DISCUSSION THREAD

**Discussion by beqramo** [mobxjs/mobx#2990] [Q&A] [answered]
Hi, I'm using Mobx in react native and I faced a problem when subscribing array to Mobx observer when I was using it with Flatlist. on array change it isn't triggered :

```tsx

import React, { useCallback } from 'react';
import { FlatList, ListRenderItemInfo } from 'react-native';
import { observer } from 'mobx-react-lite';

 const Screen = observer(() => {


  const { channelsStore } = useStores(); // Context API Store of Mobx created by MST 

  const renderItem = useCallback(
    ({ item }: ListRenderItemInfo<string>) => {
      return (
        <Text>{item}</Text>
      );
    },
    [onChatPress],
  );

  const keyExtractor = useCallback(( val) =>val, []);

  return (

        <FlatList
          data={channelsStore.channels.slice()} // without a slice it doesn't work
          renderItem={renderItem}
          keyExtractor={keyExtractor}
        />

  );
});

export default Screen

```

what is a proper way of doing it, I don't like using `slice` on every array, I tried `<Observer>` component on every possible place but nothing. 

what do you think?

**Comment by mweststrate** [comment, marked as answer]
Answering from mobile, but this is in the docs on understanding reactivity
iirc, and a few times in the issue tracker as well.

**Reply by beqramo** [reply]
Yes, I agree, I just wanted an optimal way of doing array change tracking. And one solution is above. 

I had hard time about it and I just want someone to see this as the ready to use solution.

Also other solutions are: accessing length of the array and Mobx.toJS() function.
